src/Cake.Core/IO/FileExtensions.cs

```diff
@@ -116,6 +116,28 @@ public static bool IsClrAssembly(this IFile file)
                 return false;
             }
 
+            // Is known extension?
+            switch (file.Path?.GetExtension().ToLowerInvariant())
+            {
+                case "dll":
+                case "exe":
+                case "sys":
+                case "tsp":
+                case "acm":
+                case "ax":
+                case "cpl":
+                case "drv":
+                case "efi":
+                case "mui":
+                case "ocx":
+                case "scr":
+                case null:
+                    break;
+
+                default:
+                    return false;
+            }
```

Comment:
My bad, extension includes the `.` so this should be
```suggestion
            switch (file.Path?.GetExtension().ToLowerInvariant())
            {
                case ".dll":
                case ".exe":
                case ".sys":
                case ".tsp":
                case ".acm":
                case ".ax":
                case ".cpl":
                case ".drv":
                case ".efi":
                case ".mui":
                case ".ocx":
                case ".scr":
                case null:
                    break;

                default:
                    return false;
            }
```